src/firewall/core/nftables.py

```diff
@@ -1594,14 +1594,12 @@ def _rich_rule_limit_fragment(self, limit):
             "d": "day",
         }
 
-        rate, duration = limit.value_parse()
-
         d = {
-            "rate": rate,
-            "per": rich_to_nft[duration],
+            "rate": limit.rate,
+            "per": rich_to_nft[limit.duration],
         }
 
-        burst = limit.burst_parse()
+        burst = limit.burst
```

Comment:
burst should be an integer, it needs to be parsed.

It also doesn't get normalized, like it got previously.

It would be good to add a unit test (but that is already true for the patches I had previously). Should be done.

Reply:
It's guaranteed to be an int by the `Rich_Limit` constructor, see `_burst_check()`.

Reply:
> It's guaranteed to be None or an int by the Rich_Limit constructor, see _burst_check().

Ah. I didn't know how dataclasses work. It's parsed to be an int. That's nice.

Reply:
IMO, normalizing `limit.value` is nonsensical as "normal" varies depending on firewall backend and XML. Better to breakdown `limit.value` into `limit.rate` and `limit.duration`.

For the rich language, normalization is achieved with `str(rich_limit_obj)`. But internally, if we make all these Rich_Rule objects hashable, we won't have to use this string internally at all and thus won't care about the normalized string much.

Reply:
> Ah. I didn't know how dataclasses work. It's parsed to be an int. That's nice.

Actually it's not true. The types are hints and not enforced. I changed the function to `_burst_parse()` again to make sure it's normalized to an int.